testApi/src/main/scala/africa/absa/testing/scapi/rest/response/action/AssertionResponseAction.scala

```diff
@@ -347,13 +407,11 @@ object AssertionResponseAction extends ResponseActions {
    * @param response      The response object containing the headers.
    * @return A Try[Unit] that is a Success if the "Content-Type" header value is "text/html", and a Failure with an AssertionException otherwise.
    */
-  private def assertContentTypeIsHtml(response: Response): Try[Unit] = {
-    Try {
-      assertHeaderValueEquals(response, "content-type", "text/html") match {
-        case Failure(exception) =>
-          throw AssertionException(s"Received content is not HTML type. Details: ${exception.getMessage}")
-        case _ => // Do nothing for Success
-      }
+  private def assertContentTypeIsHtml(response: Response): Try[Unit] = Try {
+    assertHeaderValueEquals(response, "content-type", "text/html") match {
+      case Failure(exception) =>
+        throw AssertionException(s"Received content is not HTML type. Details: ${exception.getMessage}")
+      case _ => // Do nothing for Success
     }
   }
```

Comment:
```suggestion
  private def assertContentTypeIsHtml(response: Response): Try[Unit] = {
    assertHeaderValueEquals(response, "content-type", "text/html").recover { f =>
      Failure(AssertionException(s"Received content is not HTML type. Details: ${f.getMessage}"))
    }
  }
```

`Failure.recover` is like `map` for exceptions. This is exactly what you need here.

Reply:
Fixed in commit 04e60c441ad8b397652ec2840da18f96300f1de6.